pkg/controller/mutators/core/reconciler.go

```diff
@@ -115,7 +115,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	// Encasing this call in a function prevents the arguments from being evaluated early.
 	id := types.MakeID(mutationObj)
 	defer func() {
-		r.reportMutator(id, ingestionStatus, startTime, conflict)
+		if !deleted {
```

Comment:
I'm not sure we want to be blind to the time it takes to process deletions, since deletion still requires some processing (e.g. removing entries from schema, re-scanning for conflicts).

If we wanted to treat them differently, maybe create an `operation` tag with values `upsert` and `delete` ?

Reply:
What do you mean by being blind to the time it takes to process deletions? Since this is executed in the defer function, I thought we'd already finished removing entries from schema and re-scanning for conflicts (assuming no error has occurred before this point).

> If we wanted to treat them differently, maybe create an operation tag with values upsert and delete?

What's the difference between that and using the `deleted` boolean directly, i.e. if `deleted` is false, it's an upsert operation, otherwise, it's a no-op.

Reply:
I think I misread what was happening here... the cache listing mutator #s makes sense to not add for deletion. We are still reporting processing times for deletion.